### PR TITLE
Improve compilation with cmake Ninja

### PR DESCRIPTION
**Pull request by romainbrenguier**
This make cudd be compiled at the configuration step, so that it is not missing when we build with Ninja.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [na] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [na] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [na] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:warning:
This PR failed Diffblue compatibility checks (cbmc commit: a70d673c3428e212df7cf01d0f2879eb9d345265).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105334919
Status will be re-evaluated on next push.
Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

  - the compatibility was already broken by an earlier merge.

File: src/solvers/CMakeLists.txt
Comment (tautschnig) [inline src/solvers/CMakeLists.txt:70]: That's a very nice addition! But then: is this really the only place where this is to be done? At least Glucose a few lines further down should see the same.

File: CMakeLists.txt
Comment (tautschnig) [inline CMakeLists.txt:56]: Hmm, how is it actually possible that CUDD is _not_ built at this time, given `download_project` above already has a `BUILD_COMMAND`? Or is the above `download_project` in fact invalid, if I read https://github.com/Crascit/DownloadProject/blob/master/DownloadProject.cmake correctly? That documentation says that `BUILD_COMMAND` and `CONFIGURE_COMMAND` are not permitted in `download_project`?!

**Review by tautschnig — APPROVED** [review]
Approving with some further comments:
1) If `download_project` in fact isn't triggering a build of CUDD, then the `download_project` invocation should be fixed in this PR, see my comments below.
2) Building CUDD at CMake configuration time does seem right to me, and also matches the spirit of our Makefile-based builds.

**Comment by tautschnig** [comment]
I think it would be good if @chrisr-diffblue or @hannes-steffenhagen-diffblue could also take a look before this one is merged.

File: CMakeLists.txt
Comment (romainbrenguier) [reply CMakeLists.txt:56]: That explains why it wasn't working then, thanks! In that case I guess we could use ExternalProject directly https://cmake.org/cmake/help/v3.10/module/ExternalProject.html instead of DownloadProject

**Review by smowton — APPROVED** [review]

File: src/solvers/CMakeLists.txt
Comment (romainbrenguier) [reply src/solvers/CMakeLists.txt:70]: done

File: CMakeLists.txt
Comment (romainbrenguier) [reply CMakeLists.txt:56]: I haven't managed to make ExternalProject work as I want (it seems `./configure` can't get executed) so I've just cleaned up the BUILD_COMMAND and CONFIGURE_COMMAND which have no effect.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: b79624836f16b11caa43180d90dc0dcdc9c60775).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/105446216

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**
